web/src/search_pill.ts

```diff
@@ -109,6 +111,90 @@ function on_pill_exit(
     $user_pill.remove();
 }
 
+function search_user_pill_data_from_term(term: NarrowTerm): SearchUserPill {
+    const emails = term.operand.split(",");
+    const users = emails.map((email) => {
+        const person = people.get_by_email(email);
+        assert(person !== undefined);
+        return person;
+    });
+    return search_user_pill_data(users, term.operator, term.negated ?? false);
+}
+
+// TODO: We're calculating `description_html` every time, even though
+// we only show it (in `generate_pills_html`) for lines with only one
+// pill. We can probably simplify things by separating out a function
+// that generates `description_html` from the information in a single
+// search pill, and remove `description_html` from the `Suggestion` type.
+export function generate_pills_html(suggestion: Suggestion): string {
+    const search_terms = Filter.parse(suggestion.search_string);
+
+    const pills: SearchPill[] = search_terms.map((term) => {
+        if (user_pill_operators.has(term.operator) && term.operand !== "") {
+            return search_user_pill_data_from_term(term);
+        }
+        return {
+            type: "search_non_user",
+            operator: term.operator,
+            operand: term.operand,
+            negated: term.negated,
+        };
+    });
+
+    const pill_render_data = pills.map((item, index) => {
```

Comment:
Is `item` the best variable name here? I'd think `pill` or` search_pill` would be clearer.

Reply:
Hmm actually should you just combine this loop with the previous one? The only block in common seems to be the `search_user` case, which the second loop does an early return for.

So I think you'd just be adding explicit `type` fields for each of the 3 return statements just below this comment? Maybe I'm missing something with how the type-checking works that would require more repetition, but I think the extra loop feels hard to read.